zitilib/actions/edge/ctrl_add_routers.go

```diff
@@ -0,0 +1,129 @@
+package edge
+
+import (
+	"errors"
+	"fmt"
+	"github.com/Jeffail/gabs"
+	"github.com/netfoundry/fablab/kernel/model"
+	"github.com/netfoundry/fablab/zitilib/actions/cli"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+)
+
+func CtrlAddRouters(regionSpec, hostSpec, componentSpec string) model.Action {
+	return &ctrlAddRouters{
+		regionSpec:    regionSpec,
+		hostSpec:      hostSpec,
+		componentSpec: componentSpec,
+	}
+}
+
+func (ar *ctrlAddRouters) Execute(m *model.Model) error {
+
+	hosts := m.GetHosts(ar.regionSpec, ar.hostSpec)
+	for _, h := range hosts {
+		components := h.GetComponents(ar.componentSpec)
+		for _, c := range components {
+			if c.Data == nil {
```

Comment:
Not sure if this is the proper use of the `Data` map. It is used to store enrollment information between creating and enrolling the edge router. Create is done from the CLI between the current fablab host and the controller. The JWT ends up on the fablab host and needs to be sent to the edge-router.